src/MagnumPlugins/BasisImporter/BasisImporter.h

```diff
@@ -0,0 +1,198 @@
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+* @brief Class @ref Magnum::Trade::BasisImporter
+*/
+
+#include <Corrade/Containers/Array.h>
+#include <Magnum/Trade/AbstractImporter.h>
+#include <Corrade/Utility/ConfigurationGroup.h>
+
+#ifndef DOXYGEN_GENERATING_OUTPUT
+#ifndef MAGNUM_BASISIMPORTER_BUILD_STATIC
+#ifdef BasisImporter_EXPORTS
+    #define MAGNUM_BASISIMPORTER_EXPORT CORRADE_VISIBILITY_EXPORT
+#else
+    #define MAGNUM_BASISIMPORTER_EXPORT CORRADE_VISIBILITY_IMPORT
+#endif
+#else
+#define MAGNUM_BASISIMPORTER_EXPORT CORRADE_VISIBILITY_STATIC
+#endif
+#define MAGNUM_BASISIMPORTER_LOCAL CORRADE_VISIBILITY_LOCAL
+#else
+#define MAGNUM_BASISIMPORTER_EXPORT
+#define MAGNUM_BASISIMPORTER_LOCAL
+#endif
+
+namespace Magnum { namespace Trade {
+
+/**
+@brief Type to transcode to
+*/
+enum class BasisTranscodingType : Int {
+    Ect1 = 0, /**< ECT1 */
+    Ect2 = 1, /**< ECT1 */
+    Bc1 = 2, /**< BC1 */
+    Bc3 = 3, /**< BC3 */
+    Bc4 = 4, /**< BC4 */
+    Bc5 = 5, /**< BC5 */
+    Bc7M6OpaqueOnly = 6, /**< BC7 M4 opaque only */
+    Pvrtc1_4OpaqueOnly = 7 /**< PVRTC1 4 opaque only */
+};
+
+/**
+@brief BASIS importer plugin
+
+Supports Basis Universal (`*.basis`) compressed images.
+
+This plugin depends on the @ref Trade and [basis_universal](https://github.com/binomialLLC/basis_universal)
+libraries and is built if `WITH_BASISIMPORTER` is enabled when building Magnum
+Plugins. To use as a dynamic plugin, you need to load the @cpp "BasisImporter" @ce
+plugin from `MAGNUM_PLUGINS_IMPORTER_DIR`. To use as a static plugin or as a
+dependency of another plugin with CMake, you need to request the `BasisImporter`
+component of the `MagnumPlugins` package and link to the `MagnumPlugins::BasisImporter`
+target. See @ref building-plugins, @ref plugins for more information.
+
+TODO: Anything else needed here? Example of how to config? See Assimp maybe
+
+# Target format
+
+Basis is a compressed format that is *transcoded* into a compressed GPU format. With
+BasisImporter, this format can be chosen in different ways:
+
+@cpp
+    /\* Choose Ect1 target format *\/
+    Containers::Pointer<AbstractImporter> importer =
+        manager.instantiate("BasisImporterEct1");
+
+    /\* Choose Bc5 target format *\/
+    Containers::Pointer<AbstractImporter> importer =
+        manager.instantiate("BasisImporterBc5");
+@ce
```

Comment:
Compiled code snippets using `@snippet`, please :) That's the first time for the plugins repo, look into `doc/snippets/` of other repos (and occurences of `@snippet` elsewhere) for how that's done. Note that it's important to do this in the root CMakeLists to have them *actually* compiled:

```cmake
# Build snippets as part of testing
if(BUILD_TESTS)
    add_subdirectory(doc/snippets)
endif()
```